pkg/providers/gcp/provider_postgressnapshot.go

```diff
@@ -129,6 +132,24 @@ func (p *PostgresSnapshotProvider) reconcilePostgresSnapshot(ctx context.Context
 		statusMessage, err := p.createPostgresSnapshot(ctx, snap, pg, config, storageClient, sqlClient)
 		return nil, statusMessage, err
 	}
+	snapshotRetention, err := str2duration.ParseDuration(string(pg.Spec.SnapshotRetention))
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to parse %q into go duration", pg.Spec.SnapshotRetention)
+		return nil, croType.StatusMessage(errMsg), errorUtil.Wrap(err, errMsg)
+	}
+	lifecycleDays := int64(math.Ceil(snapshotRetention.Hours()/24) + lifecyleAdditionalDays)
```

Comment:
Same variable as above, typo 
```suggestion
	lifecycleDays := int64(math.Ceil(snapshotRetention.Hours()/24) + lifecycleAdditionalDays)
```